app/repositories/TodoRepository.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace App\Repositories;
+
+use App\Models\Todo;
+use App\Entities\TodoEntity;
+use Illuminate\Auth\AuthManager;
+
+
+class TodoRepository
+{
+    private AuthManager $authService;
+    public function __construct(AuthManager $authService)
+    {
+        $this->authService = $authService;
+    }
+    public function getAll()
+    {
+        $todos = $this->authService->user()->todos()->get();
+        return TodoEntity::collection($todos);
+    }
+    public function createTodo(TodoEntity $todoEntity): TodoEntity
+    {
+        $todo = Todo::create([
+            'user_id' => $this->authService->user()->id,
+            'title' => $todoEntity->getTitle(),
+            'description' => $todoEntity->getDescription(),
+            'status' => $todoEntity->getStatus(),
+            'priority' => $todoEntity->getPriority(),
+            'due_date' => $todoEntity->getDueDate(),
+        ]);
+        return TodoEntity::fromModel($todo);
+    }
+
+    public function updateTodo(int $id, TodoEntity $todoEntity): TodoEntity
+    {
+        $todo = Todo::findOrFail($id);
```

Comment:
Update statements should not care about finding it first.  that should happen in the Service class.